libs/wingc/src/jsify.rs

```diff
@@ -390,17 +404,24 @@ impl<'a> JSifier<'a> {
 
 				if let Some(function_sig) = function_sig {
 					if let Some(js_override) = &function_sig.js_override {
-						let self_string = &match &callee.kind {
-							// for "loose" macros, e.g. `print()`, $self$ is the global object
-							ExprKind::Reference(Reference::Identifier(_)) => "global".to_string(),
-							ExprKind::Reference(Reference::InstanceMember { object, .. }) => {
-								self.jsify_expression(object, ctx)
+						let self_string = match callee {
+							CalleeKind::Expr(expr) => match &expr.kind {
+								// for "loose" macros, e.g. `print()`, $self$ is the global object
+								ExprKind::Reference(Reference::Identifier(_)) => "global".to_string(),
+								ExprKind::Reference(Reference::InstanceMember { object, .. }) => {
+									self.jsify_expression(&object, ctx)
+								}
+								_ => expr_string,
 							}
-
-							_ => expr_string,
+							CalleeKind::SuperCall{..} =>
+								// Note: in case of a $self$ macro override of a super call there's no clear definition of what $self$ should be,
+								// "this" is a decent option because it'll refer to the object where "super" was used, but depending on how 
+								// $self$ is used in the macro it might lead to unexpected results if $self$.some_method() is called and is
+								// defined differently in the parent class of "this".
+								"this".to_string(),
```

Comment:
We wanna figure out what `$self$` needs to be in our macro system. Imagine the case of `Array`'s `indexOf`:
the macro is `$self$.indexOf($args$)` so it'll translate to `super.indexOf(args...)`  which is good, but in case of `copy`
the macro is `Object.freeze([...($self$)])` so it'll translate to `Object.freeze([...(super)])` which is a JS error.